tests/server/test_server.py

```diff
@@ -1,16 +1,141 @@
-import sqlite3
 from contextlib import contextmanager
 from typing import Generator
 from unittest.mock import Mock
 
+import psycopg
 import pytest
 from flask.testing import FlaskClient
 
-from src.server import Server
+from src.server import QUERIES, Server
 from src.server.auth import TokenService
 from src.server.database import DatabaseHandler, DatabaseProvider, Message
 
 
+class FakeDatabase:
```

Comment:
Maybe MockDatabase would be a more accurate name?

Reply:
Imo this name is correct, because it provides fake implementation of the database.

Have a look at the explanation here: https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing